### PR TITLE
Restructure readme to sections

### PR DESCRIPTION
**Pull request by glensc**
This adds linkable section for each saver and adds overview how to use the package.

### CODE REVIEW

**Comment by glensc** [comment]
cc @krinkle @perftools/maintainers I think it looks a lot logical now.

File: README.md [hunk 1]
```diff
@@ -136,6 +99,7 @@ $config = array(
     // otherwise use any profiler that's found
     'profiler' => \Xhgui\Profiler\Profiler::PROFILER_TIDEWAYS_XHPROF,
 
+    // This allows to configure, what profiling data to capreu
```
Comment (markstory) [inline README.md]:
```suggestion
    // This allows to configure, what profiling data to capture
```

File: README.md [hunk 2]
```diff
@@ -127,7 +90,7 @@ $profiler->save($profiler_data);
 
 ## Config
 
-Here's full reference config that should give you idea what to configure.
+Here's reference config what can be configured.
```
Comment (markstory) [inline README.md]:
```suggestion
Here's a reference config of what can be configured.
```

File: README.md [hunk 3]
```diff
@@ -200,12 +133,41 @@ $config = array(
 );
 ```
 
-## Using file saver
+## Savers
+
+To deliver captured data to XHGui, you will need one of the savers to submit data to data store XHGui uses.
```
Comment (markstory) [inline README.md]:
```suggestion
To deliver captured data to XHGui, you will need one of the savers to submit to the datastore XHGui uses.
```

File: README.md [hunk 4]
```diff
@@ -217,6 +179,55 @@ To save to files, use the following configuration:
...
+        'database' => 'xhprof',
+        // Allows you to pass additional options like replicaSet to MongoClient.
+        // 'username', 'password' and 'db' (where the user is added)
+        'options' => array(),
+    ),
+```
+
+### PDO Saver
+
+PDO Saver should be able to save any PDO driver for connection.
```
Comment (markstory) [inline README.md]:
```suggestion
PDO Saver should be able to save to any PDO driver connection.
```

**Pull request merged**